geode-core/src/main/java/org/apache/geode/internal/cache/AbstractBucketRegionQueue.java

```diff
@@ -501,6 +501,18 @@ public void addToFailedBatchRemovalMessageKeys(Object key) {
     failedBatchRemovalMessageKeys.add(key);
   }
 
+  public boolean isFailedBatchRemovalMessageKeysClearedFlag() {
+    return failedBatchRemovalMessageKeysClearedFlag;
+  }
+
+  public void setFailedBatchRemovalMessageKeysClearedFlag(
+      boolean failedBatchRemovalMessageKeysClearedFlag) {
+    this.failedBatchRemovalMessageKeysClearedFlag = failedBatchRemovalMessageKeysClearedFlag;
+  }
+
+  private boolean failedBatchRemovalMessageKeysClearedFlag = false;
```

Comment:
Would you be able to move this variable to where we describe all the other variables?

Reply:
I think these new methods could be made private and have all the code that deals with failedBatchRemoveMessageKeys in this class instead of in subclasses.
For example addToFailedBatchRemovalMessageKeys could do the check isFailedBatchRemovalMessageKeysClearedFlag in this calls instead of having the caller do it.
It is okay with me if this refactoring is done in a future PR.